jdaviz/configs/specviz/tests/test_helper.py

```diff
@@ -262,6 +276,24 @@ def test_load_spectrum_list_directory(tmpdir, specviz_helper):
     assert dc_0.meta['SPORDER'] == 1
 
 
+@pytest.mark.remote_data
+def test_load_spectrum_list_directory_concat(tmpdir, specviz_helper):
+    test_data = 'https://stsci.box.com/shared/static/l2azhcqd3tvzhybdlpx2j2qlutkaro3z.zip'
+    fn = download_file(test_data, cache=True, timeout=30)
+    with ZipFile(fn, 'r') as sample_data_zip:
+        sample_data_zip.extractall(tmpdir.strpath)
+    data_path = str(tmpdir.join('NIRISS_for_parser_p0171'))
+
+    # Load two NIRISS x1d files from FITS. They have 19 and 20 EXTRACT1D
+    # extensions per file, for a total of 39 spectra to load. Also concatenate
+    # spectra common to each file into one "Combined" spectrum to load per file.
+    # Now the total is (19 EXTRACT 1D + 1 Combined) + (20 EXTRACT 1D + 1 Combined) = 41.
+    # This feature is intended for MIRI spectra, but is tested on NIRISS spectra here:
+    with pytest.warns(UserWarning, match='SRCTYPE is missing or UNKNOWN in JWST x1d loader'):
+        specviz_helper.load_spectrum(data_path, concat_by_file=True)
+    assert len(specviz_helper.app.data_collection) == 41
```

Comment:
This new test concatenates the spectra within each file into one combined spectrum per file.

Reply:
> # This feature is intended for MIRI spectra, but is tested on NIRISS spectra here

That is very confusing. Can't we have a less confusing test data?

Reply:
I understand and agree that it's confusing. I used the NIRISS spectra here only because we have them uploaded to Box and ready to go. Would it be better to find and upload some MIRI observations instead?

Reply:
Okay, I think I understand now. The key is really `concat_by_file=True` no matter what instrument. Whether it is intended for MIRI or not, there is actually no code to enforce, so the comment is actually confusing as it tries to explain the original intent but not really what is happening... I guess I can let this go if you want to leave it as-is.